errors/utils.go

```diff
@@ -0,0 +1,84 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker/utils component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+var (
+	// ErrorCodeUsingGitClone is generated when git throws an error while cloning
+	// a repository.
+	ErrorCodeUsingGitClone = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "USINGGITCLONE",
+		Message:        "Error trying to use git: %s (%s)",
+		Description:    "The specified error was thrown while using git to clone a repository",
+		HTTPStatusCode: http.StatusInternalServerError,
+	})
+
+	// ErrorCodeUsingGitCheckout is generated when git throws an error during
+	// a checkout.
+	ErrorCodeUsingGitCheckout = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "USINGGITCHECKOUT",
+		Message:        "Error trying to use git: %s (%s)",
+		Description:    "The specified error was thrown while using git checkout",
```

Comment:
The specified error refers to the variable portion %s of the error message.  If the %s was for an image name it would say the specified image name...  The description is further describing the output error message.